src/models/User.ts

```diff
@@ -0,0 +1,81 @@
+import firebase, { firestore } from 'firebase';
+import FirestoreModel from './FirestoreModel';
+import {
+  Grade,
+  ScholarshipType,
+  Gender,
+  ScholarshipAmount,
+  State,
+} from './enums';
+
+export enum Role {
+  READER,
+  CONTRIBUTOR,
+  MODERATOR,
+  ADMIN,
+}
+export enum NotificationType {
+  NONE,
+  EMAIL,
+  MOBILE,
+}
+
+interface UserData {
+  DOB?: Date;
+  schoolYear: Grade;
+  role: Role;
+  gender: Gender;
+  school?: string;
+  state?: string;
+  majors?: string[];
+  scholarshipPrefs: ScholarshipPreferences;
+  notificationSettings: NotificationSettings;
+}
+
+interface ScholarshipPreferences {
+  type: ScholarshipType[];
+  major?: string[];
+  amount: {
+    type: ScholarshipAmount;
+    min?: number;
+    max?: number;
+  };
+  schools?: string[];
+  state?: State[];
+}
+interface NotificationSettings {
+  notifications: NotificationType;
+  deadlineNotifications: NotificationType;
+  relatedNotifications: NotificationType;
+}
+
+export const converter: firestore.FirestoreDataConverter<UserData> = {
+  toFirestore: (data: UserData) => ({
+    ...data,
+  }),
+  fromFirestore: (snapshot, options) => {
+    const data = snapshot.data(options);
+    const scholarshipPrefs = data.scholarshipPrefs as ScholarshipPreferences;
+    const notificationSettings = data.notificationSettings as NotificationSettings;
+    return { ...data, scholarshipPrefs, notificationSettings } as UserData;
+  },
+};
+
+export default class User extends FirestoreModel<UserData> {
+  static get collection(): firestore.CollectionReference<UserData> {
+    return firestore().collection('users').withConverter(converter);
+  }
+
+  private constructor(id?: string, data?: UserData) {
```

Comment:
I would simply leave this as 
```suggestion
  private constructor(id: string, user?: firebase.User) {
```
We are not generating IDs for users. Similarly since we're not creating new users there is no data to pass in to save a new user. We can at most expect to read other user's data.

Reply:
Though this does make me question the design decision I made around `new Scholarship(undefined, data)` and `new Scholarship(id)`. I wonder if instead of having constructors like that if static methods like `Scholarship.new(data)` and `Scholarship.get(id)` would read better. Or maybe even having a separate `Scholarships` class and making those methods read `Scholarships.get(id)` and `Scholarships.new()` 🤔